atom/browser/api/atom_api_app.cc

```diff
@@ -262,6 +263,36 @@ v8::Local<v8::Value> App::DefaultSession(v8::Isolate* isolate) {
     return v8::Local<v8::Value>::New(isolate, default_session_);
 }
 
+bool App::OnProcessSingletonNotification(
+    const base::CommandLine& command_line,
+    const base::FilePath& current_directory) {
+  ProcessSingleton::NotificationCallback cb;
+  mate::Converter<ProcessSingleton::NotificationCallback>::FromV8(isolate(), single_instance_callback_, &cb);
+  
+  return cb.Run(command_line, current_directory);
+}
+
+bool App::MakeSingleInstance(v8::Local<v8::Function> callback) {
```

Comment:
Awesome - how do I guarantee in browser.cc or here that I'm running on the right thread? Right now I'm just flat-out calling the callback and I suspect that isn't right, especially on POSIX where the callback is probably coming back on an IO thread